### DISCUSSION TITLE
Few Qs

### DISCUSSION THREAD

**Discussion by cloudcompute** [tinacms/tinacms#2883] [General]
a. How can I place a slider or a carousel in the Home page? It is a react-based component. And other react-components like numeric slider, CTA buttons? Or, is it possible only when we pull the github content from our NextJS GraphQL API? If yes, do we need to specify some kind of placeholder while designing the content?

I am using Tina for the first time. So if someone could write a few lines about the flow, I'd appreciate.

b. Does Tina support svg files?

**Comment by cloudcompute** [comment]
Hi

I have given a quick try to Tina. But it is not production-ready. For now, I will use another CMS. Will surely keep an eye on Tina.

**Reply by perkinsjr** [reply]
I'd love to chat about what makes you think it is not production-ready. What issues you had, what we can do to improve the product, documentation, guides, etc. 

We have quite a number of projects ranging in size from personal to large-scale businesses. If you have the time I'd love to set up a time to chat?

If you are interested let me know and I can provide a calendar link where you can pick a time.

**Comment by cloudcompute** [comment]
I tried to upload 15 images using Tina and it got stuck, kept displaying the "Activity going on" icon. Then I used Cloudinary UI to upload them.  

Tina's Media Manager interface is very basic. For example, I cannot remove an image from there.

Managing the images using another UI / Image Service is  surely a deterrent for the content writers. I don't think I'd be able to convince them.

The Media Manager dialog was opened and I hit the browser's Back button. It threw the errors.

Is there a way to define the standard Header, Footer, and other UI blocks that can be applied to multiple pages.. must be there, but could not find out how?

How to insert the custom React components inside a page, like a Home page usually has an auto-image slider like component. In another page of mine, I need to include a numeric slider. 

I am actually  looking for an Headless CMS. For example, in the FAQs, I am looking forward to use the Accordion component. This means the FAQ content and the Accordion React component in which the content would be displayed should be separate. 

The Page Builder features of the different CMS that I have come across so far let us just style the basic components like Header, Text, etc. But a website page usually has other components too like Accordion, Sliders, etc. Now in this case, my styling is getting managed at two different places, one in the CMS UI and for other components I have to manage it in my own code.

My personal opinion is that if a CMS is offering the Page Builder and Preview features, then  it should let us design and show my entire Page in one shot i.e. the ability to drag and drop all widgets and style them i.e. a production-ready HTML should be there for all pages. 

Having said that, I know it takes many man-years to build such a product. Therefore, I like Headless CMS. Just write your content over there. The entire styling should be managed by our own code after pulling that content from the CMS.

Thanks and Regards

**Reply by perkinsjr** [reply]
Hey, thanks for a huge write-up, this is great feedback, I will try to answer the questions or the how to do it with TIna.

I haven't seen the issue you are running into before in regards to the Media Manager... If you could provide the errors that happened when you hit the back button that would really help the team. 

> Is there a way to define the standard Header, Footer, and other UI blocks that can be applied to multiple pages.. must be there, but could not find out how?

Tina is 100% built on the preface of you can define the schema for every block, and then the UI, and then when you insert them, they look and feel the way you want. 

What you are talking about here is known as block-based editing which can be found in our documentation here: [https://tina.io/docs/editing/blocks/](https://tina.io/docs/editing/blocks/) . You can categorize and even insert images that display to the editor, so they know exactly what they are inserting into the page. 

You can see this in the Tina Cloud Starter which is a kitchen sink style demo/starter: https://github.com/tinacms/tina-cloud-starter 

> How to insert the custom React components inside a page, like a Home page usually has an auto-image slider-like component. On another page of mine, I need to include a numeric slider.

This could be done in two ways, but because it sounds more like "Marketing" pages versus "blog" pages, using the block-based editing would be great for this, build all the different components you need and allow the content team to insert them on the page. 


> My personal opinion is that if a CMS is offering the Page Builder and Preview features, then it should let us design and show my entire Page in one shot i.e. the ability to drag and drop all widgets and style them i.e. a production-ready HTML should be there for all pages.

This is 100% possible with Tina, first, you would want to make sure you are using [Contextual Editing](https://tina.io/docs/tinacms-context/) this allows a content editor to see exactly what they are editing in "real-time". 

You would then create your schema and in your case, create a schema using the block-based building experience with every kind of custom component you would want. For example Accordion, Sliders, Hero, CTA's. Then you create your own custom React Components that can be "edited" however you want. e.g colors, content, images, etc. 

Once that is done the user can create a new page (insert any details they want, if not all of them), click on the newly created page,  and then edit / add any. This flow will be better if you are using contextual editing, where you can create a document from the contextual side, allowing a better flow. (We removed this feature to work on better flow). 

>Having said that, I know it takes many man-years to build such a product. Therefore, I like Headless CMS. Just write your content over there. The entire styling should be managed by our own code after pulling that content from the CMS.

This is exactly what we believe, the content should be driven from the headless CMS and allow you to create components, the styling, etc can be handled outside of the Headless CMS. 

I really appreciate all the feedback, this is great! Here are the links all in one spot for betting parsing:


Block Building documentation : [https://tina.io/docs/editing/blocks/](https://tina.io/docs/editing/blocks/)
Real-time UI editing: [Contextual Editing](https://tina.io/docs/tinacms-context/)
Tina Cloud Starter which is a kitchen sink style demo/starter: https://github.com/tinacms/tina-cloud-starter 

Cheers! 

James
